### DISCUSSION TITLE
How would I run some process on the server?

### DISCUSSION THREAD

**Discussion by PixelsCommander** [blitz-js/blitz#1517] [Q&A] [answered]
What is the best way to run something on the server?

My exact situation is that I want to listen to a Twitter stream (only one concurrent connection allowed) and if a mutation happens to restart the Twitter client`s singleton on the server.

Just ensuring that it runs only on the server with `typeof window !== undefined` is not enough because it also restarts on every reloads making additional connects to Twitter API, therefore, breaking API limits

**Comment by flybayer** [comment, marked as answer]
Probably you can solve that by using the same approach like we do for prisma client:

```ts
let prisma: PrismaClient

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient()
} else {
  // Ensure the prisma instance is re-used during hot-reloading
  // Otherwise, a new client will be created on every reload
  globalThis["prisma"] = globalThis["prisma"] || new PrismaClient()
  prisma = globalThis["prisma"]
}

export default prisma
```

**Reply by PixelsCommander** [reply]
Cool, thanks! I tried `global.twitterClient` but this doesn't work

**Reply by flybayer** [reply]
So you got it working, or no?

**Reply by PixelsCommander** [reply]
Yes, it is solved. Thank you.

**Comment by PixelsCommander** [comment]
@flybayer sorry for getting back to this, but part of the problem is still there. I need the logic to execute on server startup and have access to Prisma. It should not be dependent on any page being requested by user. Can you imagine any location I can use for this? Or any flow to achieve this?

**Reply by flybayer** [reply]
Unfortunately I don't really know of any way currently 😕
